tsdb/db_test.go

```diff
@@ -3489,3 +3490,225 @@ func TestDBPanicOnMmappingHeadChunk(t *testing.T) {
 
 	require.NoError(t, db.Close())
 }
+
+func TestMetadataInWAL(t *testing.T) {
+	db := newTestDB(t)
+	ctx := context.Background()
+
+	// Add some series so we can append metadata to them.
+	app := db.Appender(ctx)
+	s1 := labels.FromStrings("a", "b")
+	s2 := labels.FromStrings("c", "d")
+	s3 := labels.FromStrings("e", "f")
+	s4 := labels.FromStrings("g", "h")
+
+	app.Append(0, s1, 0, 0)
+	app.Append(0, s2, 0, 0)
+	app.Append(0, s3, 0, 0)
+	app.Append(0, s4, 0, 0)
+	app.Commit()
+
+	// Add a first round of metadata to the first three series.
+	// Re-take the Appender, as the previous Commit will have it closed.
+	app = db.Appender(ctx)
+	m1 := metadata.Metadata{Type: "gauge", Unit: "unit_1", Help: "help_1"}
+	m2 := metadata.Metadata{Type: "gauge", Unit: "unit_2", Help: "help_2"}
+	m3 := metadata.Metadata{Type: "gauge", Unit: "unit_3", Help: "help_3"}
+	app.UpdateMetadata(0, s1, m1)
+	app.UpdateMetadata(0, s2, m2)
+	app.UpdateMetadata(0, s3, m3)
```

Comment:
I realised after opening the code in text editor that we are not doing any error check for Append() Commit() UpdateMetadata(). We need to check for _all_ the calls, no error to be ignored. Maybe have a helper function inside to make it easy, like
```go
updateMetadata := func(app, s, m) {
    _, err := app.UpdateMetadata(0, s, m)
    require.NoError(t, err)
}
```

And some direct error checks.

```go
require.NoError(t, app.Commit())
```